### ISSUE TITLE
Investigate enhanced accessibility for screen reader users

### ISSUE THREAD

**Issue by PatchMixolydic** [PatchMixolydic/laria#5]
This is probably very far future, but I feel that Scala's [SCP-016: Accessible Scala][SCP-016] proposal may be beneficial for members of the [at the time of writing, hypothetical] Laria community. In short, it states that the screen reader experience should be improved by having the screen reader output match the semantics of the code and not just a bundle of symbols.

Here's an example provided in SCP-016, translated to Laria:
```rust
fn foo[A: Wibble](s: String, b: Wobble[A]) -> i32 {
```

A screen reader might read this as:
```
f n space foo left bracket capital a colon space
wibble right bracket open paren s colon space
string comma space b colon space wobble left bracket
capital a right bracket right paren hyphen greater than
i 32 left brace
```

However, the code would likely be much easier to understand if it were read as:
```
function foo
type parameter A subtypes Wibble
takes s String and b Wobble for A
and returns i 32
```

Right now, I think that this would be well-suited for integration in a language server, as that would have access to a concrete syntax tree and could generate the descriptions called for in SCP-016. Of course, we'd have to figure out how to integrate screen readers with our language server (or if that's even the optimal solution).

Of course, I'm not a screen reader user, so any feedback from those who do use one would be much appreciated.

[SCP-016]: https://github.com/scalacenter/advisoryboard/blob/b3bdcda371dc5e2ede6e35c3ff0ce99ce0fd94a1/proposals/016-verbal-descriptions.md

**Comment by PatchMixolydic** [comment]
Of course, we'd need to handle reading erroneous source code as well. Language servers ought to have good recovery support since part of their purpose is to help fix broken code, which might come in handy here. Let's take this erroneous source code:

```rust
fn foo -> i32 {
    4
}
```

This might be read as:
```
function foo
missing argument list
returns i 32
begin
return 4
end
```

**Comment by PatchMixolydic** [comment]
It's worth noting that this may be a nonissue, since just as a sighted person can see a bunch of symbols and letters and recognize it as `function foo returns 32-bit integer`, a screen-reader user may be able to hear `f n foo hyphen greater than i 32` and parse it. I've heard some screen reader users are able to turn up the dictation speed of their readers, which may help with coding efficiency, but this may be intractable for people with sensory processing disorders.
